docs/concurrency-adoption-guidelines.md

```diff
@@ -0,0 +1,179 @@
+
+```
+swift build -Xswiftc -Xfrontend -Xswiftc -warn-concurrency
+```
+
+
+#### Sendable checking today
+
+Sendable checking is currently disabled in Swift 5.5(.0) because it was causing a number of tricky situations for which we lacked the tools to resolve.
+
+Most of these issues have been resolved on today’s `main` branch of the compiler, and are expected to land in the next Swift 5.5 releases. It may be worthwhile waiting with adoption until the next version(s) after 5.5.0.
+
+For example, one of such capabilities is the ability for tuples of `Sendable` types to conform to `Sendable` as well. We recommend holding off adoption of `Sendable` until this patch lands in Swift 5.5 (which should be relatively soon). With this change, the difference between Swift 5.5 with `-warn-concurrency` enabled and Swift 6 mode should be very small, and manageable on a case by case basis.
+
+#### Backwards compatibility of declarations and “checked” Swift Concurrency
+
+Adopting Swift Concurrency will progressively cause more warnings, and eventually compile time errors in Swift 6 when sendability checks are violated, marking potentially unsafe code.
+
+It may be difficult for a library to maintain a “pre Swift 6” version along with a Swift 6 version that fully embraces the new concurrency checks. For example, it may be necessary to mark generic types as Sendable, like so:
+
+```swift
+struct Container<Value: Sendable>: Sendable { ... }
+```
+
+Here, the Value type must be marked Sendable for Swift 6’s concurrency checks to be able property work with such container. However, since the Sendable type does not exist in Swift prior to Swift 5.5, it would be difficult to maintain a library that supports both Swift 5.4+ as well as Swift 6.
+
+In such situations, it may be helpful to utilize the following trick to be able to share the same Container declaration between both Swift versions of the library:
+
+```swift
+#if compiler(>=5.5)
+public typealias MYPREFIX_Sendable = Swift.Sendable
+#else 
+public typealias MYPREFIX_Sendable = Any
+#endif
+```
+
+The Any alias is effectively a no-op when applied as generic constraint, and thus this way it is possible to keep the same Container<Value> declaration working across Swift versions.
+
+### Task Local Values and Logging
+
+The newly introduced TaskLocal Values API ([SE-0311][SE-0311]) allows for implicit carrying of metadata along with Task execution. It is a natural fit for for tracing and carrying metadata around with task execution, and e.g. including it in log messages. 
+
+We are working on adjusting Swift Log to become powerful enough to automatically pick up and log specific task local values. This change will be introduced in a source compatible way. 
+
+For now libraries should continue using logger metadata, but we expect a log of the manually passing metadata to each log statement can be replaced in the future with setting task local values. 
```

Comment:
"a log of the manually passing metadata to each log statement" needs rephrasing

Reply:
ah thanks, that's meant to be "a lot" :)